src/e_lgammaf_r.jl

```diff
@@ -0,0 +1,105 @@
+#=
+/* e_lgammaf_r.c -- float version of e_lgamma_r.c.
+* Conversion to float by Ian Lance Taylor, Cygnus Support, ian@cygnus.com.
+*/
+
+/*
+* ====================================================
+* Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+*
+* Developed at SunPro, a Sun Microsystems, Inc. business.
+* Permission to use, copy, modify, and distribute this
+* software is freely granted, provided that this notice
+* is preserved.
+* ====================================================
+*/
+=#
+
+# Matches OpenLibm behavior exactly, including return of sign
+function _logabsgamma(x::Float32)
+    hx = reinterpret(Int32, x)
+
+    #= purge off +-inf, NaN, +-0, tiny and negative arguments =#
+    signgam = 1
+    ix = hx & 0x7fffffff
+    ix ≥ 0x7f800000 && return x * x, signgam
+    ix == 0x00000000 && return Inf32, signgam
+    if ix < 0x35000000 #= |x|<2**-21, return -log(|x|) =#
+        if hx < Int32(0)
+            signgam = -1
+            return -log(-x), signgam
+        else
+            return -log(x), signgam
+        end
+    end
+    if hx < Int32(0)
+        # ix ≥ 0x4b000000 && return Inf32, signgam #= |x|>=2**23, must be -integer =#
+        t = sinpi(x)
+        t == 0.0f0 && return Inf32, signgam #= -integer =#
+        nadj = log(π / abs(t * x))
```

Comment:
how does this compare to `log(π) - log(abs(t*x))`? I think it's a free division save (where `log(π)` is precomputed)

Reply:
One could even use the irrational constant `logπ` in that case.

Reply:
To be incorporated.